cpp/models/ode_secirts/parameters_io.h

```diff
@@ -0,0 +1,1376 @@
+                            vacc_data_entry.num_vaccinations_completed;
+                    }
+                }
+                else {
+                    if (date_df == offset_first_date) {
+                        model[region_idx]
+                            .parameters.template get<DailyFullVaccinations<FP>>()[{age, SimulationDay(d)}] = 0;
+                    }
+                }
+
+                auto offset_booster_date = offset_date_by_days(date, (int)d - days_until_effective_ii);
+                if (max_date >= offset_booster_date) {
+                    if (date_df == offset_booster_date) {
+                        model[region_idx]
+                            .parameters.template get<DailyBoosterVaccinations<FP>>()[{age, SimulationDay(d)}] =
+                            vacc_data_entry.num_vaccinations_refreshed_first +
+                            vacc_data_entry.num_vaccinations_refreshed_additional;
+                    }
+                }
+                else {
+                    if (date_df == offset_first_date) {
+                        model[region_idx]
+                            .parameters.template get<DailyBoosterVaccinations<FP>>()[{age, SimulationDay(d)}] = 0;
+                    }
+                }
+            }
+        }
+    }
+    return success();
+}
+} // namespace details
+
+#ifdef MEMILIO_HAS_HDF5
+
+/**
+    * @brief Exports the time series of extrapolated real data according to
+    *   the extrapolation / approximation method used to initialize the model from
+    *   real world data.
+        (This is the vector-valued functionality of set_confirmed_cases_data())
+    * @param model vector of objects in which the data is set
+    * @param data_dir Path to transformed RKI cases files
+    * @param results_dir Path to result files
+    * @param start_date Start date of the time series to be exported.
+    * @param region vector of keys of the region of interest
+    * @param scaling_factor_inf Factor for scaling the confirmed cases to account for an estimated number of undetected cases.
+    * @param scaling_factor_icu Factor for scaling the reported ICU cases to account for possibly unreported ICU cases.
+    * @param num_days Number of days for which the time series is exported.
+    * @param divi_data_path path to divi data file
+    * @param confirmed_cases_path path to confirmed cases file
+    * @param population_data_path path to population data file
+    * @param immunity_data_path path to immunity data file
+    */
+template <class Model>
+IOResult<void> export_input_data_county_timeseries(
+    const std::vector<Model>& model, const std::string& dir, std::vector<int> const& region, Date start_date,
+    const std::vector<double>& scaling_factor_inf, double scaling_factor_icu, int num_days,
+    const std::string& divi_data_path, const std::string& confirmed_cases_path, const std::string& population_data_path,
+    const std::string& immunity_data_path)
+{
```

Comment:
This function is way too large. It reads, modifies, and writes data, all in one giant block. You can split it up into at least these three steps, maybe even more. That will allow you to actually test that the modification step works as intended, which you state in TestOdeSECIRTS.export_time_series_init is currently not possible.

Further, after splitting the function up, you don't have to add or modify any files in the test data directory, which I want to reduce as much as possible in the long run. Instead, you can use small test cases (i.e. pairs of input and expected output) to test distinct functionalities of your function. This way, the function is not only covered by a test, but actually tested.

In summary, please split up the function and rewrite the associated test(s). All of this may also apply to some of the other huge functions you use for data handling in this model.

Reply:
i already adjusted the versions of this function for the secir and secirvvs functions.  The former version was indeed horrible :) Now, we are only using tested functions here, so its less prone to errors.